### ISSUE TITLE
Switch to relative coordinates 

### ISSUE THREAD

**Issue by AntonPetrov** [RNAcentral/R2DT#131]
The [RNA 2D JSON Schema](https://github.com/LDWLab/RNA2D-data-schema) is switching from absolute to relative coordinates. This means that the code that uses R2DT JSON output needs to be updated or the SVG images that are generated using the new files will be incorrect. 

It is necessary to update R2DT without breaking the integration with [XRNA-React](https://github.com/LDWLab/XRNA-React) and [RNAcanvas](https://rnacanvas.app).

There is already a new version of Traveler that implements the change (thanks @davidhoksza!):
https://github.com/cusbg/traveler/tree/relative-labels

Here is the plan:
- I can prepare a version of R2DT that generates new JSON files
- @blakesweeney could push the updated docker image to the wwwdev API
- @pzhaojohnson and @CaedenM could test the new JSON output files using the wwwdev API and prepare to update their apps
- once everything is ready, the main R2DT API can be switched to generate new JSON files, and RNAcanvas and XRNA-React can be updated accordingly

Here are some example files provided by @davidhoksza that can be used for testing:
- [mouse_from_human_abs.colored.json](https://github.com/RNAcentral/R2DT/files/13612804/mouse_from_human_abs.colored.json)
- [mouse_from_human.colored.json](https://github.com/RNAcentral/R2DT/files/13612812/mouse_from_human.colored.json)
- [mouse_from_human.colored.svg](https://www.dropbox.com/scl/fi/4a9owm4fokbip12jkiqw9/mouse_from_human.colored.svg?rlkey=ikjrb496m074e27w9i9l0zami&dl=1)
- [mouse_from_human_abs.colored.svg](https://www.dropbox.com/scl/fi/tb7rh6ud2vjybhf1u53wu/mouse_from_human_abs.colored.svg?rlkey=2bw99f2aqkghelyionb5bt4ae&dl=1)

⚠️ This is a breaking change so other stakeholders need to be notified - hopefully they are watching this repo 🤞but I can also announce on X and add a warning on [r2dt.bio](https://r2dt.bio). 

Does this plan work? Any ideas or suggestions are very welcome!

cc @aspetr01

**Comment by davidhoksza** [comment]
Just a small comment regarding the attached files - the ```*_abs.*``` files are the original files with absolute label positions, while the non "abs" files are the ones with the new relative positioning of labels.

**Comment by AntonPetrov** [comment]
@davidhoksza or @caedenM - would you be able to comment on this please?

>Just to double-check, so it seems like residues still have absolute coordinates, but things like labels now have relative coordinates (relative to the coordinates of the corresponding residue I assume)?
>To avoid this being a breaking change, would it be possible to include some sort of "flag" property that would specify whether coordinates are to be interpreted as absolute or relative (and then to still interpret coordinates as absolute by default)?

Having a flag for absolute or relative coordinates sounds like a good idea to me - what does everyone else think?

**Comment by davidhoksza** [comment]
@pzhaojohnson Indeed, coordinates of the label and label lines relative to the coordinates of the corresponding residue

The flag should be added to the JSON schema, right? No problem with me to include it in the output from Traveler once it's in the schema...

**Comment by AntonPetrov** [comment]
@CaedenM and @aspetr01, what do you think about adding the flag suggested by Philip to the schema? 

Using a flag the transition can be made asynchronously. RNAcanvas and XRNA-React could deploy new code on their own schedule and use the flag if it's present or default to the absolute coordinates in its absence for backwards compatibility.

**Comment by pzhaojohnson** [comment]
Would using `dx` and `dy` (in place of `x` and `y`) to denote relative coordinates be a possibility?
